src/backend/distributed/master/master_modify_multiple_shards.c

```diff
@@ -196,6 +203,30 @@ master_modify_multiple_shards(PG_FUNCTION_ARGS)
 	taskList =
 		ModifyMultipleShardsTaskList(modifyQuery, prunedShardIntervalList, taskType);
 
+	/*
+	 * We should execute "TRUNCATE table_name;" on the other worker nodes before
+	 * executing the truncate commands on the shards. This is necessary to prevent
+	 * distributed deadlocks where a concurrent operation on the same table (or a
+	 * cascading table) is executed on the other nodes.
+	 *
+	 * Note that we should skip the current node to prevent a self-deadlock.
+	 */
+	if (ShouldSyncTableMetadata(relationId) && truncateOperation)
```

Comment:
minor, consider reversing arguments here. I think having truncateOperation check first makes it more readable